models/pagination.go

```diff
@@ -1,35 +1,95 @@
 package models
 
-import "strconv"
+import (
+	"math"
+	"slices"
+	"sort"
+	"strconv"
+	"strings"
+)
+
+const DefaultLimit = 20
+const DefaultSort = "default"
 
 type Filter struct {
 	Name, Value string
 }
 
 type Pagination struct {
-	limit   int
-	offset  int
-	sort    string
-	filters []Filter
+	offset           int
+	limit            int
+	sort             string
+	total            int
+	filters          []Filter
+	visiblePages     []int
+	paginationString string
 }
 
-func NewPagination(sort string, filters ...Filter) *Pagination {
-	if sort == "default" {
-		sort = ""
-	}
-
+func NewPagination(offset int, limit int, sort string, filters ...Filter) *Pagination {
 	return &Pagination{
-		sort:    sort,
-		filters: filters,
+		offset:           offset,
+		limit:            limit,
+		total:            -1,
+		sort:             sort,
+		filters:          filters,
+		visiblePages:     []int{},
+		paginationString: "0",
 	}
 }
 
+func (p *Pagination) Offset() int {
+	return p.offset
+}
+
 func (p *Pagination) Limit() int {
 	return p.limit
 }
 
-func (p *Pagination) Offset() int {
-	return p.offset
+func (p *Pagination) Total() int {
+	return p.total
+}
+
+func (p *Pagination) SetTotal(total int) {
+	p.total = total
+
+	calculatePaginationString(p)
+	calculateVisiblePages(p)
```

Comment:
Decided to have these two "cached" since there are two pagers on the folder list and these are heavier "calculations". Thoughts?

Reply:
I wouldn't bother, simplicity trumps a few calculations